datasets/xtreme_s/xtreme_s.py

```diff
@@ -0,0 +1,878 @@
+            on="path",
+        )
+
+        if split == "train":
+            df = df[(df["split"] == "train") | (df["split"] == "train_covost")]
+        else:
+            df = df[df["split"] == split]
+
+        for i, row in df.iterrows():
+            yield i, {
+                "path": os.path.join(source_path, "clips", row["path"]),
+                "audio": os.path.join(source_path, "clips", row["path"]),
+                "transcription": row["sentence"],
+                "translation": row["translation"],
+            }
+
+    # MINDS-14
+    def _minds14_split_generators(self, dl_manager):
+        archive_path = dl_manager.download_and_extract(self.config.data_urls[0])
+        audio_path = dl_manager.extract(os.path.join(archive_path, "MInDS-14", "audio.zip"))
+        text_path = dl_manager.extract(os.path.join(archive_path, "MInDS-14", "text.zip"))
+        split_paths = {k: dl_manager.download(f"minds14_splits/{k}_{self.config.lang_name}.tsv") for k in ["train", "dev", "test"]}
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "audio_path": audio_path,
+                    "text_path": os.path.join(text_path, "{}.csv".format(self.config.lang_name)),
+                    "split_tsv_file": split_paths["train"],
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "audio_path": audio_path,
+                    "text_path": os.path.join(text_path, "{}.csv".format(self.config.lang_name)),
+                    "split_tsv_file": split_paths["dev"],
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "audio_path": audio_path,
+                    "text_path": os.path.join(text_path, "{}.csv".format(self.config.lang_name)),
+                    "split_tsv_file": split_paths["test"],
+                },
+            ),
+        ]
+
+    def _minds14_generate_examples(self, audio_path, text_path, split_tsv_file):
+        key = 0
+        with open(split_tsv_file, encoding="utf-8") as split_f:
+            split_ids = set([line.split("\t")[0] for line in split_f.readlines()])
+
+        with open(text_path, encoding="utf-8") as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter=",", skipinitialspace=True)
+            next(csv_reader)
+            for row in csv_reader:
+                file_path, transcription, english_transcription, intent_class = row
+
+                # only add example if exist in splits
+                if file_path not in split_ids:
+                    continue
+
+                audio_path = os.path.join(audio_path, *file_path.split("/"))
+                yield key, {
+                    "path": audio_path,
+                    "audio": audio_path,
```

Comment:
```suggestion
                file_path = os.path.join(audio_path, *file_path.split("/"))
                yield key, {
                    "path": file_path,
                    "audio": file_path,
```

to avoid overwriting `audio_path` with a path to the wav file

Reply:
oh yeah good catch!